lib/improver/blending/weighted_blend.py

```diff
@@ -454,165 +454,216 @@ def process(self, cube, weights=None):
                         one value. Returns the original cube in this case.
 
         """
+
+        self._check_input_cube_is_iris_cube(cube)
+
+        # if the coordinate being blended is forecast_reference_time.
+        if self.coord == "forecast_reference_time":
+            self._check_all_time_points_equal(cube)
+
+        if weights is not None:
+            self._check_weights_have_coord_shape(cube, weights)
+
+        # only do blending if coordinate is a dimension coordinate
+        coord_dim = cube.coord_dims(self.coord)
+        if coord_dim:
+            result = self._perform_blending(cube, weights)
+        else:
+            # scalar coordinate
+            msg = ('Trying to blend across a scalar coordinate with only one'
+                   ' value. Returning original cube')
+            warnings.warn(msg)
+            result = cube
```

Comment:
Any need to have `_adjust_coord` in this situation? Could just do `if not coord_dim` and return early?

Reply:
I agree, this is one of the "bugs" that I referred to. If you don't have a `coord_dim`, then `_adjust_coord` will probably fail(?) - I think this is one of the advantages of simplifying the `process` method - it was hard to see this before.
(To be raised/explored in another ticket)

Reply:
Hmmm... if you alter it, will any unit or CLI tests fail?